### PR TITLE
[easy][indexer] add README

### PR DESCRIPTION
**Pull request by gegaowp**
Added a README, based on the internal Notion page:
https://www.notion.so/mystenlabs/Indexer-local-run-instructions-94c6013994584fa682ff5b3efaece323

### CODE REVIEW

File: crates/sui-indexer/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+Sui indexer is an off-fullnode service to serve data from the protocol, including both data directly generated from chain and derivative data.
+
+## today's architecture (will be changed soon)
```
Comment (randall-Mysten) [inline crates/sui-indexer/README.md:3]:
```suggestion
## Current architecture Dec 2022 (will change soon)
```

File: crates/sui-indexer/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+![indexer_simple](https://user-images.githubusercontent.com/106119108/209000367-4c7d23d8-fef2-4485-8472-89c31f0e2d62.png)
+
+## local run instructions
```
Comment (randall-Mysten) [inline crates/sui-indexer/README.md:6]:
```suggestion
## Steps to run locally
```

File: crates/sui-indexer/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+### Prerequisites
+- install local [Postgres server](https://www.postgresql.org/download/)
+- install Diesel CLI, you can follow this [guide](https://diesel.rs/guides/getting-started) until “write Rust” section
```
Comment (randall-Mysten) [inline crates/sui-indexer/README.md:9]:
```suggestion
- install Diesel CLI, you can follow the [Diesel Getting Started guide](https://diesel.rs/guides/getting-started) up to the *Write Rust* section
```

File: crates/sui-indexer/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+- [optional but handy] Postgres client like [Postico](https://eggerapps.at/postico2/), for local check, query execution etc.
+
+### Steps
+1. DB setup
+  ```
+// DB setup, run below under /sui-indexer
```
Comment (randall-Mysten) [inline crates/sui-indexer/README.md:15]:
```suggestion
// DB setup, run the following commands from the /sui-indexer folder
```

File: crates/sui-indexer/README.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+// .env file under /sui-indexer is required for diesel cmds
+// in .env file, DATABASE_URL should point to your local PG server
+// an example is:
+// DATABASE_URL=""postgres://postgres:postgres@localhost/gegao"
+diesel setup
+diesel migration run
```
Comment (randall-Mysten) [inline crates/sui-indexer/README.md:21]:
```suggestion
`diesel setup`

and then run 

`diesel migration run`
```
I couldn't suggest an update that uses ```sh as it did not render in the comment

File: crates/sui-indexer/README.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  ```
+2. cargo run under `/sui-indexer`
+  ```
```
Comment (randall-Mysten) [inline crates/sui-indexer/README.md:24]:
```suggestion
  ```sh
```

File: crates/sui-indexer/README.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  cargo run --bin sui-indexer -- --db-url "<LOCAL_DB_URL>" --rpc-client-url "https://fullnode.devnet.sui.io:443"
+  ```
+  
+### Clean up and re-run
+  ```
+  // clean up all tables in the local DB, run below until no more tables are deleted
```
Comment (randall-Mysten) [inline crates/sui-indexer/README.md:30]:
```suggestion
  // clean up all tables in the local DB, run the following command until no more tables are deleted
```

File: crates/sui-indexer/README.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+  diesel migration revert
```
Comment (randall-Mysten) [inline crates/sui-indexer/README.md:31]:
```suggestion
`diesel migration revert`
```

File: crates/sui-indexer/README.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+diesel setup
+diesel migration run
+  ```
+2. cargo run under `/sui-indexer`
+  ```
+  cargo run --bin sui-indexer -- --db-url "<LOCAL_DB_URL>" --rpc-client-url "https://fullnode.devnet.sui.io:443"
+  ```
+  
+### Clean up and re-run
+  ```
```
Comment (randall-Mysten) [inline crates/sui-indexer/README.md:29]: might be more readable if this was just text and then the command could be code format instead the whole section being a code-formatted block

Comment (gegaowp) [reply crates/sui-indexer/README.md]: good idea and changed accordingly!

**Comment by gegaowp** [comment]
@randall-Mysten thanks for the rapid review, I committed the suggested changes and the suggestion of changing rerun code block to text.

It is now ready for another review!

**Review by randall-Mysten — APPROVED** [review]
Nice, thanks for making those changes!

File: crates/sui-indexer/README.md [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+# an example is:
+# DATABASE_URL=""postgres://postgres:postgres@localhost/gegao"
+diesel setup
+
+# and then run 
+diesel migration run
+```
+2. cargo run under `/sui-indexer`
+```sh
+cargo run --bin sui-indexer -- --db-url "<LOCAL_DB_URL>" --rpc-client-url "https://fullnode.devnet.sui.io:443"
```
Comment (666lcz) [inline crates/sui-indexer/README.md:27]: Is `LOCAL_DB_URL` same as `postgres://postgres:postgres@localhost/gegao` above?

File: crates/sui-indexer/README.md [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+- [optional but handy] Postgres client like [Postico](https://eggerapps.at/postico2/), for local check, query execution etc.
+
+### Steps
+1. DB setup
+```sh
+# DB setup, run the following commands from the /sui-indexer folder
+# .env file under /sui-indexer is required for diesel cmds
+# in .env file, DATABASE_URL should point to your local PG server
+# an example is:
+# DATABASE_URL=""postgres://postgres:postgres@localhost/gegao"
```
Comment (666lcz) [inline crates/sui-indexer/README.md:19]:
```suggestion
# DATABASE_URL="postgres://postgres:postgres@localhost/gegao"
```
nit

File: crates/sui-indexer/README.md (hunk 10 above)
Comment (gegaowp) [reply crates/sui-indexer/README.md:27]: Yes, will change it to DATABASE_URL for consistency then.

**Pull request merged**